packages/core/src/http-client/http-agent.ts

```diff
@@ -77,6 +73,31 @@ const certificateOptions = (destination: Destination) => (
   }
   return options;
 };
+/**
+ * The http agents (proxy and default) use node tls for the certificate handling. This method creates the options with the pfx and passphrase. *
+ * https://nodejs.org/api/tls.html#tls_tls_createsecurecontext_options
+ * @param destination - Destination object
+ * @returns options which can be used later by tls.createSecureContext() e.g. pfx and passphrase. or {} if protocol is not https or no client information are in the defintaion.
+ * @hidden
+ */
+function getCertificateOption(destination: Destination): Record<string, any>  {
+  // http case: no certificate needed
+  if(getProtocolOrDefault(destination) === Protocol.HTTP){
+  if (destination.isTrustingAllCertificates) {
+    logger.warn('"isTrustingAllCertificates" is not available for HTTP.');
+  }
+    return {};
+  }
+  // https case: get certificate options
+  if (destination.isTrustingAllCertificates) {
```

Comment:
Without linting, nested `if-else` is terrible.

Reply:
I agree. I did execute `yarn lint:fix` but it did not fix it. Also the pipeline is happy - why is that?